src/transport_performance/gtfs/report/report_utils.py

```diff
@@ -0,0 +1,183 @@
+"""Utils to assist in the creation of a HTML report for GTFS."""
+from typing import Union
+import pathlib
+import shutil
+import os
+
+from transport_performance.utils.defence import (
+    _type_defence,
+    _handle_path_like,
+    _check_parent_dir_exists,
+)
+
+# Constant to remove non needed columns from repeated
+# pair error information.
+# This is a messy method however it is the only
+# way to ensure that the error report remains
+# dynamic and can adadpt to different tables
+# in the GTFS file.
+
+GTFS_UNNEEDED_COLUMNS = {
+    "routes": [],
+    "agency": ["agency_phone", "agency_lang"],
+    "stop_times": [
+        "stop_headsign",
+        "pickup_type",
+        "drop_off_type",
+        "shape_dist_traveled",
+        "timepoint",
+    ],
+    "stops": [
+        "wheelchair_boarding",
+        "location_type",
+        "parent_station",
+        "platform_code",
+    ],
+    "calendar_dates": [],
+    "calendar": [],
+    "trips": [
+        "trip_headsign",
+        "block_id",
+        "shape_id",
+        "wheelchair_accessible",
+    ],
+    "shapes": [],
+}
+
+
+class TemplateHTML:
+    """A class for inserting HTML string into a docstring.
```

Comment:
Should this read:
`inserting HTML string into a template` instead of docstring?